src/base/moduledownloader.lua

```diff
@@ -0,0 +1,125 @@
+--
+-- moduledownloader.lua
+-- Downloads a module from a package server
+-- Copyright (c) 2002-2016 Jason Perkins and the Premake project
+--
+
+
+---
+-- progress bar.
+--
+	local function _http_progress(total, current)
```

Comment:
I wonder if this could be added as a public HTTP function in Premake.

We also have something like this in our build scripts, but your version seems more solid, I'd rather just drop our code and use this one.